hibernate-core/src/main/java/org/hibernate/event/spi/EventType.java

```diff
@@ -76,6 +79,46 @@
 	public static final EventType<PostCollectionRemoveEventListener> POST_COLLECTION_REMOVE = create( "post-collection-remove", PostCollectionRemoveEventListener.class );
 	public static final EventType<PostCollectionUpdateEventListener> POST_COLLECTION_UPDATE = create( "post-collection-update", PostCollectionUpdateEventListener.class );
 
+	/**
+	 * Add a new event type.
+	 *
+	 * @param name - name of the custom event
+	 * @param listenerClass - the base listener class or interface associated with the entity type
+	 * @param <T> - listenerClass
+	 * @return the custom {@link EventType}
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public static synchronized <T> EventType<T> addCustomEventType(String name, Class<T> listenerClass) {
+		if ( name == null || listenerClass == null ) {
+			throw new HibernateException( "Custom EventType name and associated class must be non-null." );
+		}
+
+		final EventType eventType = EVENT_TYPE_BY_NAME_MAP.computeIfAbsent(
```

Comment:
Have we changed the implementation of `EVENT_TYPE_BY_NAME_MAP` to `ConcurrentHashMap`? If so it seems we don't need `synchronized` keyword for this method any more.